contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/schema/HiveDatabaseSchema.java

```diff
@@ -72,4 +80,56 @@ public String getTypeName() {
     return HiveStoragePluginConfig.NAME;
   }
 
+  @Override
+  public void visitTables(final RecordGenerator recordGenerator, final String schemaPath) {
+    final List<String> tableNames = Lists.newArrayList(getTableNames());
+    List<org.apache.hadoop.hive.metastore.api.Table> tables;
+    // Retries once if the first call to fetch the metadata fails
+    synchronized(mClient) {
+      try {
+        tables = mClient.getTableObjectsByName(getName(), tableNames);
```

Comment:
Can you check if this API respects user permissions of tables? If yes, can you add few tests?